ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/X11/tasks/main.yml

```diff
@@ -1,46 +1,88 @@
 #########################################################################
 # Install X11 extensions                                                #
+# x11.base.lib installp download requiring an IBMid                     #
+# https://www.ibm.com/support/pages/fileset-information-x11baselib      #
 # x11.adt.ext installp download requiring an IBMid                      #
-# https://www-01.ibm.com/support/docview.wss?uid=isg1fileset-1198156818 #
+# https://www.ibm.com/support/pages/fileset-information-x11adtext       #
 # x11.vfb installp download requiring an IBMid                          #
-# http://www-01.ibm.com/support/docview.wss?uid=isg1fileset-2054486724  #
+# https://www.ibm.com/support/pages/fileset-information-x11vfb          #
 #########################################################################
 ---
-- name: Check if X11.adt.ext is installed
-  shell: lslpp -l X11.adt.ext >/dev/null 2>&1
-  register: does_X11_adt_ext_exist
-  ignore_errors: yes
+# Rather than run lslpp - use stat on the lpp directories
+# /usr/lpp/X11.base/X11.base.lib      X11.base.lib
+# /usr/lpp/X11/include/X11/extensions X11.adt.ext
+# /usr/lpp/X11.vfb                    X11.vfb
+- name: Verify X11 software
+  block:
+  - name: Verify X11 base library
+    stat:
+      path: /usr/lpp/X11.base/X11.base.lib
+    register: _base
+  - name: Verify X11 extensions requirements
+    stat:
+      path: /usr/lpp/X11/include/X11/extensions
+    register: _ext
+  - name: Verify X11 Frame Buffer Requirements
+    stat:
+      path: /usr/lpp/X11.vfb
+    register: _vfb
+  - name: Verify Readiness #1
+    set_fact:
+      _x11rdy: "{{ _base.stat.exists
+        and _ext.stat.exists
+        and _vfb.stat.exists }}"
   tags: x11
 
-- name: Check if X11.vfb is installed
-  shell: lslpp -l X11.vfb >/dev/null 2>&1
-  register: does_x11_exist
-  ignore_errors: yes
-  tags: x11
+- name: Install from Vendor Files
+  block:
+    - name: Ensure destination for Vendor Files exists
+      file:
+        state: directory
+        path: /tmp/x11
 
-- name: Transfer and extract X11
-  unarchive:
-    src: /Vendor_Files/aix/OpenGL_X11.tar
-    dest: /tmp
-    remote_src: no
-  when: does_x11_exist.rc != 0 or does_X11_adt_ext_exist.rc != 0
-  tags: x11
+    - name: Unpack Vendor Files
+      unarchive:
+        src: /Vendor_Files/aix/OpenGL_X11.tar
+        dest: /tmp/x11
+        remote_src: no
+      ignore_errors: yes
+      register: _vendor_copied
 
-- name: Install IBM X11 Extensions - installp
-  command: installp -aXYgd /tmp/x11/X11.adt X11.adt.ext
-  ignore_errors: yes
-  when: does_X11_adt_ext_exist.rc != 0
-  tags: x11
+    - name: Install IBM X11 Extensions - installp
+      command:
+        installp -e /tmp/x11.log -aXYgd /tmp/x11 X11.adt.ext X11.vfb X11.base.lib
```

Comment:
During the install step I get the following error
```
TASK [Install IBM X11 Extensions - installp] ********************************************************************************************************************************************
task path: /Users/hkhel/AdoptOpenJDK/openjdk-infrastructure/ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/X11/tasks/main.yml:51
fatal: [p9-aix1-ojdk06.osuosl.org]: FAILED! => {"changed": true, "cmd": ["installp", "-e", "/tmp/x11.log", "-aXYgd", "/tmp/x11", "X11.adt.ext", "X11.vfb", "X11.base.lib"], "delta": "0:00:00.066162", "end": "2021-03-05
12:55:41.662029", "msg": "non-zero return code", "rc": 220, "start": "2021-03-05 12:55:41.595867", "stderr": "installp: The specified device /tmp/x11\n\tis not a valid device or file.", "stderr_lines": ["installp: The specified
device /tmp/x11", "\tis not a valid device or file."], "stdout": "", "stdout_lines": []}
...ignoring
```

I think installp does not like installing all 3 in one line?

Reply:
Same error when running on the machine directly
```
root@p9-aix1-ojdk06:[/root]installp -e /tmp/x11.log -aXYgd /tmp/x11 X11.adt.ext X11.vfb X11.base.lib
installp: The specified device /tmp/x11
        is not a valid device or file.
```

Reply:
I suspect each package's install directory/files need to be specified separately. Hence why they are separate in the role

Reply:
No, installp has no problem with multiple package names. Years of experience. This is something different - and I would need access to the Vendor_Files to examine the issue. I suspect something with the .toc file. Or, the tar file is not unpacking in /tmp/x11 - but elsewhere (e.g., /tmp/x11/x11 ?).

FYI: when .toc does not exist installp reads all files in the directory to create a .toc (table of contents)

Reply:
Under the assumption the tar contents are ./x11/FileNames I removed the creation of /tmp/x11 and renamed the destination of the unarchive: to /tmp